### PR TITLE
test: restore dev gate contracts and STT coverage

### PR DESCRIPTION
**Pull request by rmusser01**
## Summary
- reconcile stale chat, audio, Library, Watchlists, Settings, MCP, and lifecycle tests with current production ownership
- preserve narrow STT fixes for missing inputs and Parakeet empty audio while keeping real Parakeet MLX runs lazy and opt-in
- refresh the reviewed ADR-029 diagnostic inventory and isolate the test CSS cache across app mounts

## Verification
- resumed ordered suite tail: 1,994 passed, 6 skipped
- final focused affected suite: 55 passed, 13 intentional slow skips before the diagnosed CSS-cache isolation fix
- Watchlists small-to-large viewport regression: 2 passed on three consecutive runs after the fix
- diagnostic inventory: 430 owners, 1,067 TASK-492 calls, 6,631 TASK-494 calls, 4 sink files
- diagnostic architecture guard: 3 passed
- Ruff and git diff checks pass

## Review
Final code review found no Critical or Important issues. TASK-1333 is Done with all 112 acceptance criteria checked.

---
## Summary by cubic
Restores the dev test gate by aligning stale chat, audio, Library, Watchlists, Settings, MCP, and lifecycle tests with current ownership, hardens STT edge cases, and isolates the CSS cache across app mounts. Adds repair design/spec docs and refreshes the ADR-029 diagnostic inventory. Satisfies TASK-1333.

- **Bug Fixes**
  - STT: validate and reject unsafe/missing audio paths; empty Parakeet MLX input returns empty text with zero segments; preflight uses `soundfile` frames/duration.
  - Library: start export worker after recompose via `call_after_refresh` so immediate completions target the new canvas, not the outgoing form.
  - Library workspace: stronger stale-surface checks; ignore refresh results when detached/remounting or not mounted.
  - CSS: global parse cache copies rule lists to prevent cross-app mutation.
  - Benchmarks: isolate benchmark host with a private config dir (0700) and stable env; safe to reuse across runs.
  - RAG: return canonical local candidates only after prompt authority; fall back to raw pipeline context only for unsupported results.

- **Test Stability**
  - Deterministic audio capture and STT flows (mocked stream reads, VAD-independent); explicit waits replace sleeps.
  - Secure fixtures and cleanup: private dirs (0700), close DBs, isolate shared locks, use `monkeypatch.context`, skip model-catalog refresh in app boot.
  - Console/Home ownership: migrate to `pending_handoffs` with clear/ack flows; seeded provider selection to avoid state races.
  - Library/Watchlists/UI: resilient waits for mount/visibility, stable dialog visibility and focus after recomposes.
  - Diagnostic inventory refreshed to reflect current production owners.

<sup>Written for commit 6bd0bd7982af68a50f60b570185bd4d9653a443d. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/rmusser01/tldw_chatbook/pull/1154?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: tldw_chatbook/Local_Ingestion/transcription_service.py [hunk 1]
```diff
@@ -480,6 +480,9 @@ def transcribe(
         Returns:
             Dict with 'text' and 'segments' keys
         """
+        if not os.path.exists(audio_path):
+            raise TranscriptionError(f"Audio file not found: {audio_path}")
```
[the remark this answers is not in the document]
Comment (rmusser01) [reply tldw_chatbook/Local_Ingestion/transcription_service.py]: Fixed in 89fffba00. The shared local-audio entry point now calls validate_path_simple(audio_path, require_exists=True) before conversion or provider dispatch and maps validation failures to TranscriptionError. Added a regression proving unsafe paths are rejected before Parakeet model loading. Focused verification: 58 passed, 12 intentional slow skips; invalid/unsafe-path pair: 2 passed; Ruff/format/diff checks passed.

File: Tests/UI/test_product_maturity_phase3_library_contract_layout.py [hunk 2]
```diff
@@ -102,9 +102,9 @@ def test_library_source_actions_use_console_text_control_style() -> None:
     assert "$ds-library-source-action-min-width: 0;" in bundled_stylesheet
     assert "$ds-library-source-action-height: 1;" in bundled_stylesheet
     assert ".library-source-action {" in agentic_terminal
-    source_action_block = _css_block(agentic_terminal, ".library-source-action")
+    source_action_block = _css_block(agentic_terminal, ".library-source-action {")
     bundled_source_action_block = _css_block(
-        bundled_stylesheet, ".library-source-action"
+        bundled_stylesheet, ".library-source-action {"
```
[the remark this answers is not in the document]
Comment (rmusser01) [reply Tests/UI/test_product_maturity_phase3_library_contract_layout.py]: Fixed in 770357ea8. Reproduced the ValueError, restored the selector argument without the opening brace, and verified the exact CSS contract test passes.

File: Tests/Transcription/test_mlx_parakeet_transcription.py [hunk 3]
```diff
@@ -591,6 +591,7 @@ def test_result_metadata(
         assert "overlap" in result
 
 
+@pytest.mark.slow
```
[the remark this answers is not in the document]
Comment (rmusser01) [reply Tests/Transcription/test_mlx_parakeet_transcription.py:594]: Resolved in 770357ea8. The class-level slow marker remains intentional for the real-model integration suite, while the fast missing-file regression was moved into TestMLXParakeetUnit so it runs in the mandatory default suite and still proves the model loader is never called.

File: Tests/UI/test_settings_configuration_hub.py [hunk 4]
```diff
@@ -357,6 +357,32 @@ async def _wait_for_settings_input_value(
     )
 
 
+async def _wait_for_settings_select_value(
```
[the remark this answers is not in the document]
Comment (rmusser01) [reply Tests/UI/test_settings_configuration_hub.py]: Fixed in 770357ea8. Consolidated the Input and Select polling paths into one _wait_for_settings_value helper parameterized by widget type, and updated all five call sites. The three affected Settings regressions pass.

File: Tests/Transcription/test_mlx_parakeet_transcription.py [hunk 5]
```diff
@@ -590,7 +590,26 @@ def test_result_metadata(
         assert "chunk_size" in result
         assert "overlap" in result
 
+    def test_transcription_rejects_unsafe_path(self, transcription_service):
```
[the remark this answers is not in the document]
Comment (rmusser01) [reply Tests/Transcription/test_mlx_parakeet_transcription.py]: Fixed in 6bd0bd798. Moved both unsafe- and missing-path cases to the non-Darwin-gated public facade suite, parameterized across the shared boundary, and retained the assertion that provider dispatch is never called. Facade plus MLX verification: 37 passed, 11 intentional slow skips.
